analyses/tcd/metrics/tcd_metrics.R

```diff
@@ -0,0 +1,160 @@
+###############
+#### SETUP ####
+###############
+
+library(tidyverse)
+library(ggthemes)
+library(yardstick)
+library(lubridate)
+library(ggrepel)
+library(ggpubr)
+
+set.seed(1)
+
+aa_dir <- Sys.getenv("AA_DATA_DIR")
+
+metrics_dir <- file.path(
+  aa_dir,
+  "private",
+  "exploration",
+  "tcd",
+  "trigger_performance"
+)
+
+df <- read_csv(file.path(
+  aa_dir, "public", "processed", "tcd", "biomasse", "biomasse_tcd_ADM2_dekad_10.csv"
+)) %>%
+  mutate(month = ((dekad - 1) %/% 3) + 1,
+         day = 10 * ((dekad - 1) %% 3) + 1,
+         date = ymd(paste(2000, month, day, sep = "-")),
+         date_pub = date + days(20))
+
+biomasse_years <- read_csv(
+  file.path(
+    aa_dir, "public", "processed", "tcd", "biomasse", "biomasse_impact_years.csv"
+  )
+)$year
+
+list_years <- c(2001, 2004, 2009, 2011, 2017)
+
+####################
+#### PREDICTION ####
+####################
+
+pred_df <- mutate(
+  df,
+  biomasse_pred = factor(biomasse_anomaly <= 80, levels = c(T,F)),
+  bm_end_season = factor(year %in% biomasse_years, levels = c(T,F)),
+  drought_list = factor(year %in% list_years, levels = c(T,F))
+)
+
+#################
+#### METRICS ####
+#################
+
+# generate metrics for months July to December
```

Comment:
Interesting new case for the trigger report: how do we compute and communicate performance for observational triggers that can activate on a frequent basis? In Malawi, that would be daily. (In Niger that was only once.)

In this case because it's once monthly, we're treating it like a predictive trigger. Maybe there is a parallel with sudden-onset triggers here.

Just food for thought for now.

Reply:
In this case, we actually only look at it in September, not earlier, but were calculating the performance to see when we might want to set our activation timepoints. That was just a comment that accidentally was left in the code, so actually the same as Niger here. Interesting thought, though. I think we could treat it like a predictive trigger if we are doing something like monthly (so as in Niger, you could look at the individual trigger points). However, clearly impossible for daily triggers. Some interested ideas could be there for plots maybe of likelihoods across the observation period, maybe?